app/scripts/models/auth_brokers/redirect.js

```diff
@@ -59,14 +68,15 @@ define([
       });
     },
 
-    finishOAuthFlow: function (account) {
+    finishOAuthFlow: function (account, additionalResultData) {
```

Comment:
Maybe instead of only using `finishOAuthFlow`, there could be two functions `finishOAuthSignUp` and `finishOAuthSignIn`. These two functions would each call `finishOAuthFlow` with the appropriate `action`.

```js
return self.finishOAuthFlow(account, {
  action: Constants.OAUTH_ACTION_SIGNIN
}).then(function () {
```

This could be converted to:
```js
return self.finishOAuthSignIn(account)
  .then(function () {
```